scripts/upload_test_results/upload_to_github.py

```diff
@@ -0,0 +1,224 @@
+        assert upstream.exists()
+        assert upstream == empty_repo.remotes.upstream == \
+               empty_repo.remotes[remote_name]
+        upstream.fetch()
+
+        # Setup a local tracking branch of a remote branch
+
+        # create local branch "master" from remote "master"
+        empty_repo.create_head('master', upstream.refs.master)
+        # set local "master" to track remote "master
+        empty_repo.heads.master.set_tracking_branch(upstream.refs.master)
+        # checkout local "master" to working tree
+        empty_repo.heads.master.checkout()
+
+        empty_repo.create_remote(origin_name, origin_repo)
+        origin = Repo(os.path.join(repo_dir))
+        logging.info('Push new repo updates to the origin ...')
+        origin.git.push(origin_name, 'master')
+    else:
+        logging.info("Repository already created, let's update master...")
+        # Use created repo to store results
+        origin = Repo(os.path.join(repo_dir))
+        upstream = Repo(os.path.join(repo_dir))
+        origin.git.checkout('master')
+        logging.info('Pull updates from the upstream ...')
+        upstream.git.pull()
+        logging.info('Push updates to the origin ...')
+        origin.git.push(origin_name, 'master')
+
+    try:
+        # Create a new branch for daily master results
+        new_branch_name = branch_git_title + commit_name
+        origin.create_head(new_branch_name)
+        logging.info('Checkout to the new branch %s', new_branch_name)
+        origin.git.checkout(new_branch_name)
+    except Exception:
+        logging.info('Branch for the current commit exists, skip the step')
+
+    # create full git directory path where your results will be stored
+    git_dir_path = os.path.join(repo_dir, 'results', ver_folder_name)
+    logging.info('Destination path to store file in Git %s', git_dir_path)
+
+    if not os.path.exists(git_dir_path):
+        os.mkdir(git_dir_path)
+        logging.info('Path created %s', git_dir_path)
+    else:
+        logging.info('Path already existing')
+
+    return git_dir_path, new_branch_name, origin
+
+
+def add_push_to_github(git_dir_path, origin, branch_name, commit_msg,
+                       title_msg, github_user, target_repo, github_token):
+    # Add your test results to the repo, commit them and then push them
+    # to create PR on Github
+    for j in glob.glob("{}/*.xml".format(file_path)):
+        file_name = os.path.basename((j))
+        logging.info("File name %s", file_name)
+
+        # Return path to its initial status
+        source_file_path = argument.res_dir
+        source_file_path = os.path.join(source_file_path, file_name)
+        logging.info('Source directory file path %s', source_file_path)
+        git_file_path = os.path.join(git_dir_path, file_name)
+        logging.info('Destination path to file to commit %s', git_file_path)
+        shutil.copy(source_file_path, git_file_path)
```

Comment:
An error will be raised here while the file already exists. For example, we both run sanitycheck with `iotdk`, so you may have already uploaded `iotdk.xml` when I try to upload it.

Reply:
@wangnuannuan Yes, it will. I can add exception, but that question we can raise during TG Meeting today. Because, we except each company will upload results of their boards, and nobody can duplicate its test results. For example, your company is responsible for iotdk, and your must check results and provide only results for iotdk. No any other developer or company can upload iotdk results and duplicate them. Please, don't forget to raise that question during TW Meeting.

Reply:
@nashif Do you have any input about that? Each company uploads separate boards, right?